### DISCUSSION TITLE
Creates a brax system from a MuJoCo model (Custom environment)

### DISCUSSION THREAD

**Discussion by amdee** [google/brax#325] [General]
Hi, I am trying to recreate/create a custom environment using a MuJoCo model as shown in the [v2 READMe.md](https://github.com/google/brax/tree/main/brax/v2l) especially unitree a1 robot. My approach was as follows;

1. I created a *UnitreeA1* class to represent the robot (it includes path to mjcf file). It has the reset and step method.
2. Registered the class through the "__init__.py" file in the envs folder 
3. Updated the base.py and mjcf.py files to account for cylindrical shapes.
4. I am running into the _**"RuntimeError: unrecognized collider pair: (, )" error**_. I know what causes the _RuntimeError_. Do I need to implement methods for a contact of a cylinder with other types of geometric shapes? if so, can you please provide guidance how to go about that?
5. Is possible to provide guidance on how to recreate the unitree a1 example on the _[v2 README.md](https://github.com/google/brax/tree/main/brax/v2)_

Thanks for your help! I am super excited about Brax (especially v2). Once I can recreate the custom environments, I would like to contribute to the documentation.

**Comment by btaba** [comment]
Hi @amdee thanks for using brax! If the runtime error is because you added a cylindrical shape to brax on your branch, then you would either have to [1] turn off collisions for that shape using `contype` `conaffinity` in the XML, or [2] implement the collision pair that is throwing the error. To implement the collision, here's the code reference for where that happens:
https://github.com/google/brax/blob/1d5fbd7cdc71708e8549e34a0aa485e7dc8a460c/brax/geometry/contact.py#L446

@erikfrey for the a1 example in the README

**Reply by amdee** [reply]
@erikfrey would you mind providing direction on how to reproduce unitree A1 example in the README? Please

**Comment by amdee** [comment]
@btaba Thanks for replying. I ended up approximating the cylindrical shape with a Capsule but at one point I ended wanted to implement the collision pair for the cylindrical shape. I will revisit the ideal, thanks for the info.

**Comment by erikfrey** [comment]
Hello @amdee !  Here's a few steps to replicate the A1 example:

* Make sure you're using brax at HEAD (we posted a few mjcf fixes not yet in pypi)
* You need to tweak a few things in a1.xml
  * delete `cone="eliptic"` from the options (brax only supports pyramidal friction)
  * delete the extra `solimp` from the foot (it's in our TODO to support multiple distinct sol params per env)
  * simplify collision by setting collision default geom conaffinity to "0" (brax is very performance sensitive to number of collisions in a physics step)

Here's a gist of those three changes:

https://gist.github.com/erikfrey/21517b337772ad011b2d8593a7e8409d

There's more you can do to optimize the env for training perf, but hopefully that gets you started.

**Reply by amdee** [reply]
@erikfrey Thanks.
